app/models/package_manager/maven.rb

```diff
@@ -148,5 +163,40 @@ def self.project_name(name)
         artifactId: sections[1]
       }
     end
+
+    def self.get_pom(group_id, artifact_id, version, seen=[])
+      xml = get_xml(
+        "http://repo1.maven.org/maven2/#{group_id}/#{artifact_id}/#{version}/#{artifact_id}-#{version}.pom"
+      )
+
+      seen << [group_id, artifact_id, version]
+
+      next_group_id = xml.locate('distributionManagement/relocation/groupId/?[0]').first || group_id
+      next_artifact_id = xml.locate('distributionManagement/relocation/artifactId/?[0]').first || artifact_id
+      next_version = xml.locate('distributionManagement/relocation/version/?[0]').first || version
+
+      if seen.include?([next_group_id, next_artifact_id, next_version])
+        xml
+
+      else
+        begin
+          get_pom(next_group_id, next_artifact_id, next_version, seen)
+        rescue Faraday::Error
+          xml
+        end
+      end
+    end
+
+    def self.licenses(xml)
+      xml_licenses = xml
+        .locate('licenses/license/name')
+        .flat_map(&:nodes)
+      return xml_licenses if xml_licenses.any?
+
+      comments = xml.locate('*/^Comment')
```

Comment:
Does this actually work? I didn't actually get a chance to test it in my recommendation 😄

Reply:
The test passes!